openwisp_radius/api/views.py

```diff
@@ -172,3 +178,52 @@ def get_user(self, serializer, *args, **kwargs):
 
 
 obtain_auth_token = csrf_exempt(ObtainAuthTokenView.as_view())
+
+
+class PasswordChangeView(BasePasswordChangeView):
+    authentication_classes = (BaseTokenAuthentication,)
+
+
+password_change = PasswordChangeView.as_view()
+
+
+class PasswordResetView(BasePasswordResetView):
+
+    def get_serializer_context(self):
+        organization = Organization.objects.get(slug=self.kwargs['slug'])
+        default_url = settings.PASSWORD_RESET_URLS.get('default')
+        password_reset_url = settings.PASSWORD_RESET_URLS.get(str(organization.pk), default_url)
+        context = {
+            'request': self.request,
+            'organization': self.kwargs['slug'],
+            'password_reset_url': password_reset_url
+        }
+        return context
+
+    def dispatch(self, *args, **kwargs):
+        try:
+            organization = Organization.objects.get(slug=kwargs['slug'])
+            if self.request.POST.get('email', None):
+                email = self.request.POST['email']
+                try:
+                    user = User.objects.get(email=email)
+                    if (organization.pk,) not in user.organizations_pk:
+                        message = _('Email "{}" is not member '
+                                    'of "{}"').format(email, kwargs['slug'])
+                        logger.warning(message)
+                        return HttpResponseForbidden(message)
+                except User.DoesNotExist:
+                    raise Http404()
+        except Organization.DoesNotExist:
```

Comment:
there's some duplication with `DispatchOrgMixin`, try to reuse that as much as possible,
also try to avoid duplication of:

```python
if (self.organization.pk,) not in user.organizations_pk:
    message = _('User "{}" is not member '
                'of "{}"').format(user.username, kwargs['slug'])
    logger.warning(message)
```

Better create a method in `DispatchOrgMixin` that does that operation and reuse it.
Duplication is evil because it adds maintenance overhead on the project.
Remember that software has to be maintained and updated once is written, the more duplication we have, the more code we will have to maintain and update.